lib/node_modules/@stdlib/time/base/parse-duration/lib/main.js

```diff
@@ -0,0 +1,79 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// VARIABLES //
+
+var RE_TIME_UNIT = /(\d+)([a-z]+)/gi;
+
+
+// MAIN //
+
+/**
+* Parses a duration string such as `1m3s10ms` into an object with `days`, `hours`, `minutes`, `seconds`, and `milliseconds`.
+*
+* @param {string} str - duration string
+* @returns {Object} duration object
+*
+* @example
+* var obj = parseDuration( '1m3s10ms' );
+* // returns { 'days': 0, 'hours': 0, 'minutes': 1, 'seconds': 3, 'milliseconds': 10 }
+*
+* @example
+* var obj = parseDuration( '1m3s' );
+* // returns { 'days': 0, 'hours': 0, 'minutes': 1, 'seconds': 3, 'milliseconds': 0 }
+*/
+function parseDuration( str ) {
+	var parts;
+	var out;
+	var i;
+
+	out = {
+		'days': 0,
+		'hours': 0,
+		'minutes': 0,
+		'seconds': 0,
+		'milliseconds': 0
+	};
+	if ( str.length === 0 ) {
+		return out;
+	}
+	parts = RE_TIME_UNIT.exec( str );
+	while ( parts ) {
+		i = parts[ 2 ];
+		if ( i === 'd' || i === 'D' ) {
+			out.days = parseInt( parts[ 1 ], 10 );
+		} else if ( i === 'h' || i === 'H' ) {
+			out.hours = parseInt( parts[ 1 ], 10 );
+		} else if ( i === 'm' || i === 'M' ) {
+			out.minutes = parseInt( parts[ 1 ], 10 );
+		} else if ( i === 's' || i === 'S' ) {
+			out.seconds = parseInt( parts[ 1 ], 10 );
+		} else if ( i === 'ms' || i === 'MS' ) {
+			out.milliseconds = parseInt( parts[ 1 ], 10 );
+		}
+		parts = RE_TIME_UNIT.exec( str );
```

Comment:
Am I correct in thinking that once `exec` returns `null`, the regexp index property resets to `0`? Meaning, we are fine using a regexp from the parent scope and not creating it fresh each time the function is invoked?

Reply:
Yes, this is the case.